tests/network/sriov.go

```diff
@@ -647,6 +646,40 @@ func checkInterfacesInGuest(vmi *v1.VirtualMachineInstance, interfaces []string)
 	}
 }
 
+// isInterfaceOnRootPCIComplex checks whether device is on root complex
+// Count the number of occurrences of PCI addresses along the device path
+// If only one is found, it indicates that the device is on the root complex,
+// otherwise a PCI bridge address will be listed along the path as the parent
+// of this device.
+// Examples:
+// on root       /sys/devices/pci0000:00/0000:00:03.0/net/eth1
+// not on root   /sys/devices/pci0000:00/0000:00:02.7/0000:08:00.0/net/eth1
+func isInterfaceOnRootPCIComplex(vmi *v1.VirtualMachineInstance, iface string) (bool, error) {
+	ifacePath, err := console.RunCommandAndStoreOutput(vmi,
+		fmt.Sprintf("find /sys/devices/ -name %s", iface), 15*time.Second)
+	if err != nil {
+		return false, err
+	}
+	ifacePathSlice := strings.Split(ifacePath, "/")
+	const busTokenIndex = 4
+	if len(ifacePathSlice) < busTokenIndex {
+		return false, fmt.Errorf("interface path %s is not as expected", ifacePath)
+	}
+	ifacePathSlice = ifacePathSlice[busTokenIndex-1:]
+
+	return countNumberOfPCIAddressesAlongPath(ifacePathSlice) == 1, nil
```

Comment:
Is it possible to compare to the actual path to the expected root PCI complex?
I find it clearer then using regex.

Reply:
Isn't it `/sys/devices/pci0000:00`?

Reply:
> That would have been nice but we don't know the PCI addresses and frankly shouldn't care

On regular interfaces we can specify the pci address in the guest, but I do not know/remember if it works for sriov/hostdevices.

> Isn't it `/sys/devices/pci0000:00`?

In the doc examples, both start from root.
Am I missing something?